crates/hotshot/src/traits/election/static_committee.rs

```diff
@@ -4,78 +4,128 @@
 // You should have received a copy of the MIT License
 // along with the HotShot repository. If not, see <https://mit-license.org/>.
 
-use std::{marker::PhantomData, num::NonZeroU64};
+use std::collections::BTreeMap;
+use std::num::NonZeroU64;
 
 use ethereum_types::U256;
-// use ark_bls12_381::Parameters as Param381;
-use hotshot_types::traits::signature_key::StakeTableEntryType;
 use hotshot_types::{
-    signature_key::BLSPubKey,
     traits::{
-        election::Membership, network::Topic, node_implementation::NodeType,
-        signature_key::SignatureKey,
+        election::Membership,
+        network::Topic,
+        node_implementation::NodeType,
+        signature_key::{SignatureKey, StakeTableEntryType},
     },
     PeerConfig,
 };
 #[cfg(feature = "randomized-leader-election")]
 use rand::{rngs::StdRng, Rng};
-use tracing::debug;
-
-/// Dummy implementation of [`Membership`]
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
-pub struct GeneralStaticCommittee<T, PUBKEY: SignatureKey> {
-    /// All the nodes participating and their stake
-    all_nodes_with_stake: Vec<PUBKEY::StakeTableEntry>,
-    /// The nodes on the static committee and their stake
-    committee_nodes_with_stake: Vec<PUBKEY::StakeTableEntry>,
-    /// builder nodes
-    committee_nodes_without_stake: Vec<PUBKEY>,
-    /// the number of fixed leader for gpuvid
+
+/// The static committee election
+pub struct GeneralStaticCommittee<T: NodeType> {
+    /// The nodes eligible for leadership.
+    /// NOTE: This is currently a hack because the DA leader needs to be the quurm
+    /// leader but without voting rights.
+    eligible_leaders: Vec<<T::SignatureKey as SignatureKey>::StakeTableEntry>,
+
+    /// The nodes on the committee and their stake
+    stake_table: Vec<<T::SignatureKey as SignatureKey>::StakeTableEntry>,
+
+    /// The nodes on the committee and their stake, indexed by public key
+    indexed_stake_table:
+        BTreeMap<T::SignatureKey, <T::SignatureKey as SignatureKey>::StakeTableEntry>,
```

Comment:
These are mostly renamed, but I added an index by key which we didn't have before